src/bun.js/node/node_fs_watcher.zig

```diff
@@ -403,6 +404,32 @@ pub const FSWatcher = struct {
 
                             current_task.append(relative_path, event_type, .destroy);
 
+                            const full_path = bun.default_allocator.dupeZ(u8, path_slice) catch unreachable;
```

Comment:
The file will often already be watched. That means this allocation will usually be unnecessary which hurts performance. Instead, this should be copied into a temporary buffer of size `bun.MAX_PATH_BYTES + 1` and only cloned if needed. That will also fix the memory leak this code creates where `full_path` is not freed

Reply:
This branch is executed when the event's kind is `.directory`. I think this occurs only when files are created, renamed, or deleted in a directory. Perhaps I misunderstood, but in most cases, this branch is only executed once.

https://github.com/oven-sh/bun/blob/13b5d9d4de4e9fe897d373a6ed7fdf2c9884c71f/src/bun.js/node/node_fs_watcher.zig#L366

Reply:
I tried resuing `_on_file_update_path_buf`. However, when a new file is created, it needs a heap-allocated `file_path`. I deferred this operation to `addFile` by using the `copy_file_path` parameter.